### PR TITLE
DOC: Adding offsets to API ref

### LINKED ISSUE #24436: DOC: Add offsets methods to the API reference
The methods of classes in `pandas.tseries.offsets` are not in the API reference. Not sure if there is a reason for it, or it just needs to be implemented.

We have the classes added to:
https://pandas-docs.github.io/pandas-docs-travis/api.html#date-offsets

But most of the attributes and methods are not in `api.rst`, so their pages are not generated. See for example:
https://pandas-docs.github.io/pandas-docs-travis/generated/pandas.tseries.offsets.MonthEnd.html#pandas.tseries.offsets.MonthEnd

Can they simply be added? Or is there a reason not to, or it's more complex than it looks like?

CC: @jbrockmendel @mroeschke

### PR DESCRIPTION
**Pull request by datapythonista**
- [x] closes #24436

Adding Date Offsets API documentation. Added in the same format as other sections. Detected what are properties and what are methods with the `callable` function. The order of the classes is the one already used. The order inside properties and methods is alphabetical.

@jreback In `timeseries.rst` we already have a table with the different offsets: https://pandas-docs.github.io/pandas-docs-travis/timeseries.html#dateoffset-objects Not sure what would you like to add more.

This is how it looks like:
offsets

CC: @mroeschke @jbrockmendel

### CODE REVIEW

**Comment by mroeschke** [comment]
Is it possible to give `Properties` and `Methods` a different (smaller) heading style than the offset class? It's a bit difficult to pick out the offset from those two headings.

**Comment by datapythonista** [comment]
Totally agree, but I think the short answer is no.

I copied the format of: https://pandas-docs.github.io/pandas-docs-travis/api.html#scalars

IMO, what we should do is split the API page in several sections (see the sections in the "API Reference" block in https://pandas-dev.github.io/pandas-sphinx-theme/pr-datapythonista_base/, but possibly with all Index classes in the same section). Then we should have at least one extra level of title/subtitle..., and we'll be able to do it.

Will open an issue soon, and see if there is agreement.

**Comment by jreback** [comment]
> @jreback In timeseries.rst we already have a table with the different offsets: https://pandas-docs.github.io/pandas-docs-travis/timeseries.html#dateoffset-objects Not sure what would you like to add more.

it didn’t have clickable links before

**Comment by datapythonista** [comment]
ah, I see, thanks for clarifying. But that's somehow unrelated, the class pages already existed, it's just that they are not links.

I'll add those in a separate PR.

**Comment by datapythonista** [comment]
Created #24448 for the links.

File: doc/source/api.rst [hunk 1]
```diff
@@ -2139,54 +2139,1370 @@ Date Offsets
...
+    DateOffset.rule_code
+
+Methods
+~~~~~~~
+.. autosummary::
+   :toctree: generated/
+
+    DateOffset.apply
+    DateOffset.copy
+    DateOffset.isAnchored
```
Comment (jreback) [inline doc/source/api.rst]: I think we are removing / not-advertising ``isAnchored``

File: doc/source/api.rst [hunk 2]
```diff
@@ -2139,54 +2139,1370 @@ Date Offsets
...
+
+Properties
+~~~~~~~~~~
+.. autosummary::
+   :toctree: generated/
+
+    DateOffset.freqstr
+    DateOffset.kwds
+    DateOffset.name
+    DateOffset.nanos
```
Comment (jreback) [inline doc/source/api.rst]: I think we consider .nanos private @jbrockmendel ?

**Review by jreback — CHANGES_REQUESTED** [review]
can you auto-generate these?

**Comment by datapythonista** [comment]
I "autogenerated" them, would you prefer to generate the content of `api.rst` dynamically? If we do it for date offsets, it probably makes sense to do it for everything.

Regarding the private methods, sphinx will warn (and eventually we'll make it fail for warnings). So it'd probably be a better option to just rename them to `_isAnchored` (or `_is_anchored`) and `_nanos`. Is there any reason to not do that?

Related to this PR, I created #24451 to split `api.rst` in multiple files.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
this is fine for now (and ok on the splitting).

**Pull request merged**